airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.server.handlers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airbyte.api.model.generated.ConnectorBuilderProjectIdWithWorkspaceId;
+import io.airbyte.api.model.generated.ConnectorBuilderProjectWithWorkspaceId;
+import io.airbyte.api.model.generated.ExistingConnectorBuilderProjectWithWorkspaceId;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.ConnectorBuilderProject;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.validation.json.JsonValidationException;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Singleton
+public class ConnectorBuilderProjectsHandler {
+
+  private final ConfigRepository configRepository;
+  private final Supplier<UUID> uuidSupplier;
+
+  @Inject
+  public ConnectorBuilderProjectsHandler(final ConfigRepository configRepository,
+                                         final Supplier<UUID> uuidSupplier) {
+    this.configRepository = configRepository;
+    this.uuidSupplier = uuidSupplier;
+  }
+
+  // This should be deleted when cloud is migrated to micronaut
+  @Deprecated(forRemoval = true)
+  public ConnectorBuilderProjectsHandler(final ConfigRepository configRepository) {
+    this.configRepository = configRepository;
+    this.uuidSupplier = UUID::randomUUID;
+  }
+
+  private ConnectorBuilderProject builderProjectFromUpdate(final ExistingConnectorBuilderProjectWithWorkspaceId projectCreate) {
+    return new ConnectorBuilderProject().withBuilderProjectId(projectCreate.getBuilderProjectId()).withWorkspaceId(projectCreate.getWorkspaceId())
+        .withName(projectCreate.getBuilderProject().getName())
+        .withManifestDraft(new ObjectMapper().valueToTree(projectCreate.getBuilderProject().getDraftManifest()));
+  }
+
+  private ConnectorBuilderProject builderProjectFromCreate(final ConnectorBuilderProjectWithWorkspaceId projectCreate) {
+    final UUID id = uuidSupplier.get();
+
+    return new ConnectorBuilderProject().withBuilderProjectId(id).withWorkspaceId(projectCreate.getWorkspaceId())
+        .withName(projectCreate.getBuilderProject().getName())
+        .withManifestDraft(new ObjectMapper().valueToTree(projectCreate.getBuilderProject().getDraftManifest()));
```

Comment:
Small note: this currently isn't respecting the `hasDraft` value of the `ConnectorBuilderProjectWithWorkspaceId` object that is passed into this function.

I tried calling this with `hasDraft` set to `false` and let `draftManifest` unset, but when I then called the `/list` endpoint, it returned `hasDraft: true` for this project, because this function is always setting the `manifestDraft` so something regardless of the `hasDraft` value.

Thinking about this more, the request body for this `/create` API endpoint probably shouldn't take in a `hasDraft` field at all, since that just meant to be an informative field when listing. So related to my above comment, I think `hasDraft` and `builderProjectId` should be removed from the `/create` request body.

And this method should be updated to check if the `draftManifest` is null on the `projectCreate` argument, and if so it should not set it on the database record.

Reply:
You are right, `hasDraft` is only relevant for the response, not for the request. I will separate these two cases in the other PR https://github.com/airbytehq/airbyte-platform/pull/46 (it also fixes the issue with `draftManifest` always being set to something even if you unset it)